evalml/automl/automl_search.py

```diff
@@ -631,6 +635,28 @@ def __init__(
         if not engine:
             self._engine = SequentialEngine()
         else:
+            valid_engines = [
+                "cf_threaded",
+                "cf_process",
+                "dask_threaded",
+                "dask_process",
+            ]
+            if isinstance(engine, str):
+                if engine in valid_engines:
```

Comment:
My favorite way to do this kind of "switch-like" statement in python is to use a dictionary lookup:

```python
engines = {"cf_threaded": CFEngine(CFClient(ThreadPoolExecutor())), .... }
engine = engines.get(engine)
if not engine:
    raise ValueError(...)
```

I prefer it this way because I think we can get rid of all of the if statements here. The dict lookup will implicitly verify the engine is a valid string.

Reply:
Sorry, I went off half-cocked.  I used a dictionary but a try/except with KeyError!  Should be better, though.